src/ops/create.rs

```diff
@@ -61,42 +86,10 @@ pub fn create_hashes<W>(path: &PathBuf, ignored_files: BTreeSet<String>, algo: A
             }
 
             unreachable!();
-        })
-        .collect();
+        }));
 
     pb.show_tick = false;
     pb.tick();
     pb.finish_print("");
     hashes
 }
-
-fn create_hashes_p(hashes: &mut BTreeMap<String, CpuFuture<String>>, path: &PathBuf, ignored_files: &BTreeSet<String>, prefix: String, pool: &CpuPool,
-                   algo: Algorithm, remaining_depth: DepthSetting, follow_symlinks: bool) {
-    for file in fs::read_dir(&path).unwrap().map(Result::unwrap) {
-        let file_type = file.file_type().unwrap();
-        let file_name_s = prefix.clone() + file.file_name().to_str().unwrap();
-        let ignored = ignored_files.contains(&file_name_s);
-
-        let mut subpath = path.clone();
-        subpath.push(file.path());
-
-        if file_type.is_file() {
-            let hash = if ignored {
-                // TODO: ideally, this'd be a futures::done() but I was unable to generalise it to do both CpuFuture and Future
```

Comment:
Because it's no longer a TODO. Ignored files write directly into the hashes BTreeMap, they skip the hashes_f BTreeMap entirely.